### PR TITLE
WPF initial improvements

### PR DESCRIPTION
**Pull request by nmilcoff**
Hey folks, thanks for all the amazing work you've done so far. I really enjoyed working on this PR.

Summary of my changes:
- Update Windows solution to load all necessary projects
- Update breaking changes for WPF
- Installed Newtonsoft from NuGet 
- Update naming convention in some cases (e.g. `IImageWrapper` -> `IImage`)
- Add new converters `FigmaViewPropertySetter` and `FigmaCodePropertyConverter` (most of the code is still commented out) 
- Update shape and basic converters for WPF: Focus on Line, Text, Ellipse, Rectangle so far

You can check this also using the `BasicRendering` sample, [the Figma file used is public](https://www.figma.com/file/qVIUSmz8AkK158q8pY3L23).

### CODE REVIEW

**Comment by netonjm** [comment]
Thank you very much @nmilcoff ! could you please include some screenshot about how look the rendering?

File: FigmaSharp/FigmaSharp.Wpf/ViewWrappers/Image.cs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+				//image.Width = value.Width;
+				//image.Height = value.Height;
+			}
+		}
+
+		protected ImageSource image;
+        public Image(ImageSource image)
+        {
+            this.image = image;
+        }
```
Comment (netonjm) [inline FigmaSharp/FigmaSharp.Wpf/ViewWrappers/Image.cs:28]: styling seems to be broken here

File: FigmaSharp/FigmaSharp.Wpf/ViewWrappers/ImageView.cs [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+        IImage image;
+        public IImage Image 
+        {
+            get => image;
+            set
+            {
+                image = value;
+                var nativeImage = (ImageSource)Image.NativeObject;
+                imageView.Source = nativeImage;
```
Comment (netonjm) [inline FigmaSharp/FigmaSharp.Wpf/ViewWrappers/ImageView.cs:30]: 👍

File: FigmaSharp/FigmaSharp.Wpf/Views/CanvasImage.cs [hunk 3]
```diff
@@ -34,14 +34,15 @@ namespace FigmaSharp.Wpf
 {
     public class CanvasImage : Canvas, IDisposable
     {
-        public Image ImageView {
+        public System.Windows.Controls.Image ImageView
```
Comment (netonjm) [inline FigmaSharp/FigmaSharp.Wpf/Views/CanvasImage.cs:37]: why expose public in abstraction wpf native types?

Comment (nmilcoff) [reply FigmaSharp/FigmaSharp.Wpf/Views/CanvasImage.cs:37]: True, I'll make it a private field

File: FigmaSharp/FigmaSharp.Wpf/ViewWrappers/Image.cs (hunk 1 above)
Comment (nmilcoff) [reply FigmaSharp/FigmaSharp.Wpf/ViewWrappers/Image.cs:32]: Thanks, it's fixed now

**Comment by nmilcoff** [comment]
Thanks for reviewing this and for the feedback @netonjm! Here's a screenshot for everyone to appreciate my art skills 😂 :

imagen

File: FigmaSharp/FigmaSharp/FigmaSharp.csproj [hunk 4]
```diff
@@ -21,9 +21,7 @@
     <Folder Include="Properties\" />
   </ItemGroup>
   <ItemGroup>
-    <PackageReference Include="Newtonsoft.Json" Version="12.0.1">
-        <Private>False</Private>        
-    </PackageReference>
```
Comment (netonjm) [inline FigmaSharp/FigmaSharp/FigmaSharp.csproj:26]: Let's keep this PackageReference as private it's shared between all the toolkits

File: FigmaSharp/FigmaSharp/FigmaSharp.csproj [hunk 5]
```diff
@@ -21,9 +21,7 @@
     <Folder Include="Properties\" />
   </ItemGroup>
   <ItemGroup>
-    <PackageReference Include="Newtonsoft.Json" Version="12.0.1">
-        <Private>False</Private>        
-    </PackageReference>
+    <PackageReference Include="Newtonsoft.Json" Version="12.0.1" />
```
Comment (netonjm) [inline FigmaSharp/FigmaSharp/FigmaSharp.csproj:24]: Why this change?

File: FigmaSharp/FigmaSharp/FigmaSharp.csproj (hunk 4 above)
Comment (netonjm) [inline FigmaSharp/FigmaSharp/FigmaSharp.csproj:26]: Why this change?

File: samples/FigmaSharp/BasicRendering/BasicRendering.Wpf/BasicRendering.Wpf.csproj [hunk 6]
```diff
@@ -34,9 +34,13 @@
     <ErrorReport>prompt</ErrorReport>
     <WarningLevel>4</WarningLevel>
   </PropertyGroup>
+  <PropertyGroup>
+    <StartupObject>
+    </StartupObject>
+  </PropertyGroup>
   <ItemGroup>
     <Reference Include="Newtonsoft.Json, Version=12.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed, processorArchitecture=MSIL">
```
Comment (netonjm) [inline samples/FigmaSharp/BasicRendering/BasicRendering.Wpf/BasicRendering.Wpf.csproj:42]: maybe we could move to package references

File: FigmaSharp/FigmaSharp/FigmaSharp.csproj (hunk 4 above)
Comment (nmilcoff) [reply FigmaSharp/FigmaSharp/FigmaSharp.csproj:26]: This shouldn't have changed 🤦  will commit in a sec

File: samples/FigmaSharp/BasicRendering/BasicRendering.Wpf/BasicRendering.Wpf.csproj (hunk 6 above)
Comment (nmilcoff) [reply samples/FigmaSharp/BasicRendering/BasicRendering.Wpf/BasicRendering.Wpf.csproj:42]: On my way!

File: FigmaSharp/FigmaSharp.Wpf/Converters/FigmaElipseConverter.cs [hunk 7]
```diff
@@ -1,29 +1,29 @@
...
 using FigmaSharp.Models;
+using FigmaSharp.Views;
+using FigmaSharp.Services;
+
 namespace FigmaSharp.Wpf.Converters
 {
     public class FigmaElipseConverter : FigmaElipseConverterBase
     {
-        public override IViewWrapper ConvertTo(FigmaNode currentNode, ProcessedNode parent)
+        public override Type GetControlType(FigmaNode currentNode) => typeof(View);
```
Comment (netonjm) [inline FigmaSharp/FigmaSharp.Wpf/Converters/FigmaElipseConverter.cs:11]: this override should return the native view to convert in this case I suppose typeof(CanvasImage)

File: FigmaSharp/FigmaSharp.Wpf/Converters/FigmaFrameEntityConverter.cs [hunk 8]
```diff
@@ -29,24 +29,61 @@
...
-
-            return figmaImageView;
-        }
-
-        public override string ConvertToCode(FigmaNode currentNode)
-        {
-            return string.Empty;
-        }
+    {
+        public override Type GetControlType(FigmaNode currentNode) => typeof(ImageView);
```
Comment (netonjm) [inline FigmaSharp/FigmaSharp.Wpf/Converters/FigmaFrameEntityConverter.cs:41]: mm... we need some changes in the base. GetControlType including FigmaRendererService as a parameter.
for now do a return here a typeof (FigmaSharp.Wpf.View) , I will fix it asap.

File: FigmaSharp/FigmaSharp.Wpf/Converters/FigmaLineConverter.cs [hunk 9]
```diff
@@ -27,26 +27,31 @@
...
+using FigmaSharp.Views;
+using FigmaSharp.Services;
+using System;
+
 namespace FigmaSharp.Wpf.Converters
 {
     public class FigmaLineConverter : FigmaLineConverterBase
     {
-        public override IViewWrapper ConvertTo(FigmaNode currentNode, ProcessedNode parent)
+        public override Type GetControlType(FigmaNode currentNode) => typeof(View);
```
Comment (netonjm) [inline FigmaSharp/FigmaSharp.Wpf/Converters/FigmaLineConverter.cs:39]: return here a typeof (CanvasImage)

File: FigmaSharp/FigmaSharp.Wpf/Converters/FigmaRectangleVectorConverter.cs [hunk 10]
```diff
@@ -26,30 +26,33 @@
...
-
-    public class FigmaRectangleVectorConverter : FigmaRectangleVectorConverterBase
-    {
-        public override IViewWrapper ConvertTo(FigmaNode currentNode, ProcessedNode parent)
-        {
-            var figmaEntity = (FigmaRectangleVector)currentNode;
+{ 
+    public class FigmaRectangleVectorConverter : RectangleVectorConverterBase
+    {
+        public override Type GetControlType(FigmaNode currentNode) => typeof(View);
```
Comment (netonjm) [inline FigmaSharp/FigmaSharp.Wpf/Converters/FigmaRectangleVectorConverter.cs:39]: return here a typeof (CanvasImage)

File: FigmaSharp/FigmaSharp.Wpf/Converters/FigmaRegularPolygonConverter.cs [hunk 11]
```diff
@@ -9,17 +11,19 @@
 namespace FigmaSharp.Wpf.Converters
 {
     public class FigmaRegularPolygonConverter : FigmaRegularPolygonConverterBase
-    {
-        public override IViewWrapper ConvertTo(FigmaNode currentNode, ProcessedNode parent)
-        {
+    {
+        public override Type GetControlType(FigmaNode currentNode) => typeof(ImageView);
```
Comment (netonjm) [inline FigmaSharp/FigmaSharp.Wpf/Converters/FigmaRegularPolygonConverter.cs:15]: return here a typeof (CanvasImage)

File: FigmaSharp/FigmaSharp.Wpf/Converters/FigmaVectorEntityConverter.cs [hunk 12]
```diff
@@ -28,22 +28,27 @@
...
+using FigmaSharp.Views;
+using FigmaSharp.Services;
+using System;
 
 namespace FigmaSharp.Wpf.Converters
 {
     public class FigmaVectorEntityConverter : FigmaVectorEntityConverterBase
     {
-        public override IViewWrapper ConvertTo(FigmaNode currentNode, ProcessedNode parent)
+        public override Type GetControlType(FigmaNode currentNode) => typeof(ImageView);
```
Comment (netonjm) [inline FigmaSharp/FigmaSharp.Wpf/Converters/FigmaVectorEntityConverter.cs:39]: return here a typeof (CanvasImage)

File: FigmaSharp/FigmaSharp.Wpf/Converters/FigmaVectorViewConverter.cs [hunk 13]
```diff
@@ -28,22 +28,26 @@
...
 using FigmaSharp.Models;
+using FigmaSharp.Services;
+using FigmaSharp.Views;
 
 namespace FigmaSharp.Wpf.Converters
 {
     public class FigmaVectorViewConverter : FigmaVectorViewConverterBase
     {
-        public override IViewWrapper ConvertTo(FigmaNode currentNode, ProcessedNode parent)
+        public override Type GetControlType(FigmaNode currentNode) => typeof(ImageView);
```
Comment (netonjm) [inline FigmaSharp/FigmaSharp.Wpf/Converters/FigmaVectorViewConverter.cs:38]: return here a typeof (CanvasImage)

File: FigmaSharp/FigmaSharp/FigmaModels.cs [hunk 14]
```diff
@@ -475,6 +475,7 @@ public virtual bool HasImage ()
 		[Category ("Stroke")]
         [DisplayName ("Strokes")]
         public FigmaPaint[] strokes { get; set; }
+        public bool HasStrokes => strokes?.Length > 0;
```
Comment (netonjm) [inline FigmaSharp/FigmaSharp/FigmaModels.cs:478]: 👍

File: FigmaSharp/FigmaSharp.Wpf/Converters/FigmaElipseConverter.cs (hunk 7 above)
Comment (nmilcoff) [reply FigmaSharp/FigmaSharp.Wpf/Converters/FigmaElipseConverter.cs:11]: Should it be `CanvasImage` or `FrameworkElement`? Also I see the Cocoa version is returning `NSView`, should that be changed to `NSImageView` there? ([ref](https://github.com/netonjm/FigmaSharp/blob/master/FigmaSharp/FigmaSharp.Cocoa/Converters/FigmaElipseConverter.cs#L44))

Comment (netonjm) [reply FigmaSharp/FigmaSharp.Wpf/Converters/FigmaElipseConverter.cs:11]: Cocoa allows custom draws over NSView, here you should change it to the same native view are you going to create  to draw that Elipse

**Comment by nmilcoff** [comment]
@netonjm thanks for the feedback! I applied the requested changes :) I expect my upcoming PRs can / will be smaller and easier to review

**Comment by hbons** [comment]
👏 thanks @nmilcoff!

**Comment by netonjm** [comment]
Great work @nmilcoff ! 💯
